Created progress bars and participant page

JacobWestfield opened the pull request.

Legrank commented on src/components/progressBar.jsx:
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const ProgressBar = ({ value, color, name }) => {
```
функция выбора цвета есть смысл вынести в отдельный файл, она же нам будет нужна ещё как минимум в кнопке

Legrank commented on src/components/progressBar.jsx:
```diff
@@ -0,0 +1,44 @@
+
+const ProgressBar = ({ value, color, name }) => {
+    const pickColor = (color) => {
+        switch (color) {
```
функция выбора цвета есть смысл вынести в отдельный файл, она же нам будет нужна ещё как минимум в кнопке

Legrank commented on src/components/participantPage.jsx:
```diff
@@ -0,0 +1,74 @@
+                        <div className="md-4">
+                            <h4>Прогресс изучения технологий:</h4>
+                            <ProgressBar
+                                name="HTML"
```
Добавление ProgressBar сделать циклом. Проходимся по объекту user.progress, ключи идут в пропс name, значение в пропс value
Не обязательно у всех будут именно эти 3 скила, могут быть другие.

JacobWestfield replied:
Не получится циклом. нужно другое решение, там нужно выбирать цвет по заданию

JacobWestfield replied:
разве что сделать проверку на технологию и от нее цвет

Legrank replied:
Я вижу 2 варианта.
1.Простой. Сделать список технология-цвет, если технологии нету в списке выбирать случайный цвет.
2.Сложнее. Каждый раз выбирать случайный цвет, но цвета должны быть разными.(если css красный, то другие технологии  не могут быть красными)

Legrank commented on src/components/participantPage.jsx:
```diff
@@ -0,0 +1,74 @@
+                            {Object.keys(user.social).map((key) => {
+                                return (
+                                    <a key={key} href={user.social[key]}>
+                                        <h3>{key}</h3>
```
Идея для улучшения: создать компонент который в зависимости от названия приложения будет добавлять значек перед ссылкой.

Legrank commented on src/components/participantPage.jsx:
```diff
@@ -0,0 +1,74 @@
+
+                    <div>
+                        <h1>
+                            <span className={`badge bg-${user.badgeColor}`}>
+                                Teamlead
+                            </span>
```
У узера есть поле badge, выводить информацию в зависимости от этого поля.

JacobWestfield committed:
добавил иконки, поменял цикл bars, добавил утилиты

JacobWestfield committed:
Пофиксил ошибки

JacobWestfield committed:
updated

JacobWestfield committed:
updated

Legrank commented on src/utils/pickColor.js:
```diff
@@ -0,0 +1,19 @@
+ * @param {string} technology name of technology (HTML, CSS...)
+ * @returns string className
+ */
+export const pickColor = (technology) => {
```
Идея утилитных функций в том что они должны быть максимально универсальными что бы их можно было использовать в разных местах. Сейчас мы не можем использовать эту функцию в кнопке для выбора цвета, т.к. у кнопки нету технологий. Логику связи технология-цвет логичнее вынести в прогресс бар. Он получает название технологии и выбирает для неё какой-то цвет. Дальше используем pickColor что бы превратить цвет в класс.

Legrank commented on src/components/participantPage.jsx:
```diff
@@ -0,0 +1,72 @@
+
+          <div>
+            <h1>
+              <span className={`badge bg-${user.badgeColor}`}>
+                {user.badge}
+              </span>
```
Будем ли мы обрабатывать ситуацию что у человека больше 1 бейджа?

JacobWestfield replied:
На твое усмотрение, скорее всего тимлид и так разработчик подразумевается,так что мне кажется одного бейджа хватит

JacobWestfield committed:
Merge branch 'main' into progressbar-participantPage

JacobWestfield committed:
Reusable pickColor + new func in participantPage

Сделал pickColor.js переиспользуемым, добавил конфиг в компонент страницы участника для выбора цвета в зависимости от технологии чтобы выполнялось задание хакатона

JacobWestfield committed:
new colors config

Добавил рандомный цвет для progressBar если технологии нет в стандартных
